### PR TITLE
feat: added new `service_name` parameter for the `exec`, `request` and `wait` instructions. NOTE: the previous methods' signature will be maintained during a deprecation period, we suggest users update the methods' calls to this new signature.

### PR DESCRIPTION
**Pull request by leoporoli**
Context here: https://www.notion.so/kurtosistech/Design-Embedding-wait-in-the-add_service-instruction-79cc8089f00d4b6a96e31611cfa3a500#e5361bfb861b4d5c81b31c7f0cfb4e7f

### CODE REVIEW

File: core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/kurtosis_plan_instruction_base_test.go [hunk 1]
```diff
@@ -15,6 +15,12 @@ type KurtosisPlanInstructionBaseTest interface {
 	// GetStarlarkCode should return the serialized starlark code matching the argument dict from GetExpectedArguments
 	GetStarlarkCode() string
 
+	// GetStarlarkCodeForAssertion sometimes, for instance when testing positional args, we need a starlark
+	//code for execution (which will be provided by GetStarlarkCode) and another version for assertion
+	//this method provide the second one,
+	//this can be used only if having different scripts (one for execution and another for assertion) is needed, otherwise this should return an empty string
+	GetStarlarkCodeForAssertion() string
```
Comment (gbouv) [inline core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/kurtosis_plan_instruction_base_test.go:22]: Sweet! 👍 
nit: blank space on each line right after `//`

File: core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/request_framework_test_case_1.go [hunk 2]
```diff
@@ -17,42 +17,42 @@ import (
 )
 
 const (
-	request_serviceName = service.ServiceName("web-server")
-	request_portId      = "port_id"
-	request_method      = "GET"
-	request_contentType = ""
-	request_endpoint    = "/"
-	request_body        = ""
+	requestServiceName = service.ServiceName("web-server")
```
Comment (gbouv) [inline core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/request_framework_test_case_1.go]: nit: If you want to remove `_` maybe prefix with `requestTestCase1` or whatever, here we're under the impression it's `requestServiceName` as the service name of the request.

**Review by gbouv — APPROVED** [review]
Looks good! I like the StarlarkCodeForAssertion!

**Comment by gbouv** [comment]
Meta point on this PR, it's doing a lot. You're modifying exec, request and wait, you're updating GetHttpRequest and PostHttprequest, you're adding this cool GetStarlarkCodeForAssertions for testing. I think I didn't forget anything? I believe each of those things could have lived in its own PR, almost. Typically:
1. Start by making serviceName optional for GetHttpRequest
2. Make serviceName optional for PostHttpRequest
3. Add GetStarlarkCodeForAssertions for tests, maybe start adding some of the tests
4. Update exec to add ServiceName to it
5. Update request to add ServiceName to it
6. Update wait to add ServiceName to it

Splitting PRs not only make the reviewers work easier, but it also encourages him to reflect at a higher level on the change you're making. Like here adding `GetStarlarkCodeForAssertions` in the middle of an already huge PR, the reviewer is tempted to just give up with a "yes, why not". If you make `GetStarlarkCodeForAssertions` the ONLY change in a PR, the reviewer will be much more thoughtful on the core change you're making, asking himself "Does this GetStarlarkCodeForAssertions makes sense at a high level, will it be useful generally or just for a very specific usecase, is there an alternative approach I can think of"

Curious to hear your thoughts and if you disagree happy to chat more about it :) Those things really matters to me, but I also don't want to be the annoying person rambling alwasy rambling about his things, so definitely definitely let me know if you think I'm wrong here and hopefully we can find a middle ground that fits all of us nicely

**Comment by gbouv** [comment]
That being said, I appreciate sharing the notion doc with contexts in the PR description :)

**Comment by leoporoli** [comment]
> Meta point on this PR, it's doing a lot. You're modifying exec, request and wait, you're updating GetHttpRequest and PostHttprequest, you're adding this cool GetStarlarkCodeForAssertions for testing. I think I didn't forget anything? I believe each of those things could have lived in its own PR, almost. Typically:
> 
> 1. Start by making serviceName optional for GetHttpRequest
> 2. Make serviceName optional for PostHttpRequest
> 3. Add GetStarlarkCodeForAssertions for tests, maybe start adding some of the tests
> 4. Update exec to add ServiceName to it
> 5. Update request to add ServiceName to it
> 6. Update wait to add ServiceName to it
> 
> Splitting PRs not only make the reviewers work easier, but it also encourages him to reflect at a higher level on the change you're making. Like here adding `GetStarlarkCodeForAssertions` in the middle of an already huge PR, the reviewer is tempted to just give up with a "yes, why not". If you make `GetStarlarkCodeForAssertions` the ONLY change in a PR, the reviewer will be much more thoughtful on the core change you're making, asking himself "Does this GetStarlarkCodeForAssertions makes sense at a high level, will it be useful generally or just for a very specific usecase, is there an alternative approach I can think of"
> 
> Curious to hear your thoughts and if you disagree happy to chat more about it :) Those things really matters to me, but I also don't want to be the annoying person rambling alwasy rambling about his things, so definitely definitely let me know if you think I'm wrong here and hopefully we can find a middle ground that fits all of us nicely

@gbouv  I agree that this PR includes a lot and I agree it would have been better to plan small PRs instead, the problem was that when I started I thought that this will be a small one, and at the beginning I thought that the `exec` and the `wait` instructions needed the refactor, then I realized that the `request` instructions also needed, so it was the first increment, then I realize that I was going to need the `GetStarlarkCodeForAssertions` and I add it, then when I create the PR yesterday was very late in the day when I realize that it was big, and I had to run, so I decided to send it anyway because this way it could be reviewed early in the morning, but yes I know that this decision complicates the review for you. I'll try to fix this for next PRs

**Comment by gbouv** [comment]
Yup yup, all good Leandro no worries at all. Just wanted to mention higher level thoughts on this PR, aside from the code looking good!
If you were sharing the same feeling when you submitted this PR, then just ignore my message :)

**Pull request merged**